test/unit/lib/plugins/package/lib/packageService.test.js

```diff
@@ -611,28 +614,176 @@ describe('lib/plugins/package/lib/packageService.test.js', () => {
     });
 
-    it('should ignore `package.artifact` if `functions[].package.individually', () => {
+    it.skip('should ignore `package.artifact` if `functions[].package.individually', () => {
       // Confirm that fnIndividual was packaged independently
       //
       // Replace
       // https://github.com/serverless/serverless/blob/b12d565ea0ad588445fb120e049db157afc7bf37/test/unit/lib/plugins/package/lib/packageService.test.js#L262-L287
     });
 
-    it('should support `functions[].package.artifact`', () => {
+    it.skip('should support `functions[].package.artifact`', () => {
       // Confirm that file pointed at `functions.fnArtifact.package.artifact` is configured as function level artifact
     });
+
+    describe('with absolute artifact path', () => {
+      let absoluteArtifactFilePath;
+      let zipContent;
+
+      before(async () => {
+        const zip = new JsZip();
+        zip.file('index.js', '');
+        zipContent = await zip.generateAsync({ type: 'base64' });
+        absoluteArtifactFilePath = path.join(process.cwd(), 'newArtifact.zip');
+        await fs.promises.writeFile(absoluteArtifactFilePath, zipContent);
+      });
+
+      describe('while deploying whole service', () => {
+        const s3UploadStub = sinon.stub();
+        const awsRequestStubMap = {
+          Lambda: {
+            getFunction: {
+              Configuration: {
+                LastModified: '2020-05-20T15:34:16.494+0000',
+              },
+            },
+          },
+          S3: {
+            upload: s3UploadStub,
+            listObjectsV2: {},
+          },
+          CloudFormation: {
+            describeStacks: {},
+            describeStackResource: { StackResourceDetail: { PhysicalResourceId: 'resource-id' } },
+          },
+          STS: {
+            getCallerIdentity: {
+              ResponseMetadata: { RequestId: 'ffffffff-ffff-ffff-ffff-ffffffffffff' },
+              UserId: 'XXXXXXXXXXXXXXXXXXXXX',
+              Account: '999999999999',
+              Arn: 'arn:aws:iam::999999999999:user/test',
+            },
+          },
+        };
+
+        beforeEach(() => {
+          s3UploadStub.resetHistory();
+        });
+
+        it('for function', async () => {
+          await runServerless({
+            fixture: 'function',
+            cliArgs: ['deploy'],
+            lastLifecycleHookName: 'aws:deploy:deploy:uploadArtifacts',
+            awsRequestStubMap,
+            configExt: {
+              functions: {
+                other: {
+                  package: {
+                    artifact: absoluteArtifactFilePath,
+                  },
+                },
+              },
+            },
+          });
+
+          // One with CF template, one with function artifact and one with provided function artifact
+          expect(s3UploadStub).to.be.calledThrice;
```

Comment:
Do you think it's important to confirm on that (?) I guess there are some unrelated uploads involved there

Reply:
I wanted to differentiate this way between the two cases (for service-wide scenario there's one upload less)

Reply:
It looks as implementation detail, which may change without breaking functionality (e.g. in some other unrelated part we'll do S3 upload and it'll break tests here), so i wouldn't cover this.

Reply:
makes sense, removed :+1: